File: 2003/IntroToVBNETPart2/Tutorial/Count2.vb
Imports System
Imports Microsoft.VisualBasic 

Public Class Count

	Const EOF As Integer = -1

	Shared Sub Main ()
		Dim i, nwhite, nother As Integer
		Dim c As Char
		Dim ndigit(10) As Integer

		nwhite = 0 : nother = 0
		For i = 0 To 10
			ndigit(i) = 0
		Next i

		While True
			i = Console.Read ()
			If i = EOF Then Exit While
			c = Chr ( i )
			Select Case c
			Case "0"C To "9"C
				ndigit ( Asc(c) - Asc("0"C) ) += 1 
			Case " "C, ControlChars.Lf, ControlChars.Tab
				nwhite += 1 
			Case Else
				nother += 1 
			End Select
		End While
		Console.Write ( "digits =" ) 
		For i = 0 To 9
			Console.Write ( " {0}", ndigit(i) ) 
		Next i
		Console.WriteLine ( vbCrLf & "white space = {0}, other = {1}", _
			nwhite, nother ) 
	End Sub
End Class

File: 2003/IntroToVBNETPart2/Tutorial/Count2.cs
using System ;

public class Count
{
	const int EOF = -1 ;

	public static void Main ()
	{
		int i, nwhite, nother ;
		char c ;
		int [] ndigit = new int [10] ;

		nwhite = nother = 0 ;
		for ( i = 0 ; i < 10 ; ++i )
			ndigit[i] = 0 ;

		while ( ( i = Console.Read () ) != EOF )
		{
			c = (char) i ;
			switch ( c )
			{
			case '0':
			case '1':
			case '2':
			case '3':
			case '4':
			case '5':
			case '6':
			case '7':
			case '8':
			case '9':
				ndigit [ c - '0' ]++ ;
				break ;
			case ' ':
			case '\n':
			case '\t':
				nwhite++ ;
				break ;
			default:
				nother++ ;
				break ;
			}
		}
		Console.Write ( "digits =" ) ;
		for ( i = 0 ; i < 10 ; ++i )
			Console.Write ( " {0}", ndigit[i] ) ;
		Console.WriteLine ( "\nwhite space = {0}, other = {1}", 
			nwhite, nother ) ;
	}
}